source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,14 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint endpoint(endpoint_.value());
```

Comment:
You can avoid copying here by just adding a non-const value() function here: https://github.com/lyft/envoy/blob/master/include/envoy/common/optional.h#L37

The method signature should be: ```T& value()``` and the body will be the same as the other one.